atomicapp/utils.py

```diff
@@ -20,6 +20,20 @@
 def isTrue(val):
     return str(val).lower() in true_values
 
+def addShelveKV(dbname, key, val):
+    with closing(shelve.open(dbname, writeback = True)) as s:
+        s[key] = val
+
+def appendShelveKV(dbname, key, val):
+    with closing(shelve.open(dbname, writeback = True)) as s:
+        temp = s[key]
+        temp.append(val)
```

Comment:
in shelve you have to do this. you need to fetch the entire value and  append and save the entire object back again.